tests/unit/test_config.py

```diff
@@ -0,0 +1,104 @@
+"""Tests for the `Config` classes."""
+
+from typing import Callable
+from unittest.mock import MagicMock, patch
+
+import toml
+
+from autoimport.config import AutoImportConfig, Config
+
+
+class TestConfig:
+    """Tests for the `Config` class."""
+
+    def test_get_valid_option(self) -> None:
+        """
+        Given: a `Config` instance with a `config_dict` populated,
+        When a value is retrieved for an existing option,
+        Then the value of the option is returned
+        """
+        config_dict = {"foo": "bar"}
+        config = Config(config_dict=config_dict)
+
+        result = config.get_option("foo")
+
+        assert result == "bar"
+
+    def test_get_value_for_missing_option(self) -> None:
+        """
+        Given: a `Config` instance with a `config_dict` populated,
+        When: a value is retrieved for a option not defined in the `config_dict`,
+        Then: `None` is returned
+        """
+        config_dict = {"foo": "bar"}
+        config = Config(config_dict=config_dict)
+
+        result = config.get_option("baz")
+
+        assert result is None
+
+    def test_get_value_for_no_config_dict(self) -> None:
+        """
+        Given: a `Config` instance without a given `config_dict`,
+        When: a value is retrieved for an option,
+        Then: `None` is returned
+        """
+        config = Config()
+
+        result = config.get_option("foo")
+
+        assert result is None
+
+
+class TestAutoImportConfig:
+    """Tests for the `AutoImportConfig`."""
+
+    @patch("autoimport.config.get_pyproject_path")
+    def test_valid_pyproject(
+        self, mock_get_pyproject_path: MagicMock, create_tmp_file: Callable
```

Comment:
I'm not sure I understand what you mean but I'll do my best.

The idea would be that in the setup phase of the tests, you'd create the directory and file structure similar of what you'll encounter in a real life environment. Very much like you do with the `create_tmp_file`. 

If you make the `AutoImportConfig` class accept the starting path as the first argument, you can pass it to `_find_config` and then to `get_pyproject_path` instead of `Path.cwd()`

Reply:
Yep it looks marvelous, doesn't it? :)

Reply:
Now that you agreed to change it, maybe it makes sense to remove the patches also in `TestGetPyprojectPath` too

Reply:
addressed in 6934c55